### PR TITLE
feat: add latest issues card component with hooks and tests

### PR DESCRIPTION
**Pull request by rakshett**
## Fixes 

https://issues.redhat.com/browsex/KFLUXUI-745

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2025-10-17 at 2 11 51 PM
## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

## Summary by CodeRabbit

* **New Features**
  * Added a "Latest issues" card showing up to 10 active issues for the current namespace with severity icons, titles, descriptions, formatted detection timestamps, loading/error/empty states, and a footer link to view all issues; supports optional styling.

* **Tests**
  * Added comprehensive tests covering loading, error, empty, success, severity variants, timestamps, accessibility, and edge cases.

### CODE REVIEW

File: src/components/Issues/LatestIssuesCard.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+const getSeverityIcon = (severity: Issue['severity']) => {
+  const IconComponent = SEVERITY_ICONS[severity] || InfoCircleIcon;
+  return <IconComponent className={`latest-issues-card__icon--${severity}`} />;
+};
+
+interface LatestIssuesCardProps {
+  className?: string;
+}
+
+const formatTimestamp = (dateString: string) => {
```
Comment (rrosatti) [inline src/components/Issues/LatestIssuesCard.tsx]: nit: can we use some of the datetime utils located in `src/shared/components/timestamp/datetime.ts`?

File: src/components/Issues/LatestIssuesCard.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                      className="latest-issues-card__title-row"
+                    >
+                      <FlexItem className="latest-issues-card__icon-container">
+                        {getSeverityIcon(issue.severity)}
+                      </FlexItem>
+                      <FlexItem flex={{ default: 'flex_1' }}>
+                        <div className="latest-issues-card__title">{issue.title}</div>
+                      </FlexItem>
+                    </Flex>
+                    <div className="latest-issues-card__description">{issue.description}</div>
```
Comment (rrosatti) [inline src/components/Issues/LatestIssuesCard.tsx]: can we use `Text` component from `@patternfly/react-core` lib here?

File: src/types/issues.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+export interface IssueScope {
```
[the remark this answers is not in the document]
Comment (rakshett) [reply src/types/issues.ts]: removed the issues type and used the interface from the kite/issue-type.ts

File: src/components/Issues/LatestIssuesCard.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import { ArrowRightIcon } from '@patternfly/react-icons/dist/esm/icons/arrow-right-icon';
+import { ExclamationCircleIcon } from '@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons/dist/esm/icons/exclamation-triangle-icon';
+import { InfoCircleIcon } from '@patternfly/react-icons/dist/esm/icons/info-circle-icon';
+import { useLatestIssues } from '~/hooks/useIssues';
+import { getErrorState } from '~/shared/utils/error-utils';
+import { Issue } from '~/types/issues';
+import { useNamespace } from '../../shared/providers/Namespace';
+import './LatestIssuesCard.scss';
+
```
[the remark this answers is not in the document]
Comment (rakshett) [reply src/components/Issues/LatestIssuesCard.tsx]: Removed and used the icons by referring the 439 PR

File: src/hooks/useIssues.ts
[the remark this answers is not in the document]
Comment (rakshett) [reply src/hooks/useIssues.ts]: Yes, the branch wasn't updated and so, these changes where hsown. Now the branch is updated and those changes are removed

File: src/components/Issues/LatestIssuesCard.tsx (hunk 1 above)
Comment (rakshett) [reply src/components/Issues/LatestIssuesCard.tsx]: Now using `dateTimeFormatter` and `isValid` from `~/shared/components/timestamp/datetime`

File: src/components/Issues/LatestIssuesCard.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                      </FlexItem>
+                      <FlexItem flex={{ default: 'flex_1' }}>
+                        <div className="latest-issues-card__title">{issue.title}</div>
+                      </FlexItem>
+                    </Flex>
+                    <div className="latest-issues-card__description">{issue.description}</div>
+                    <div className="latest-issues-card__timestamp">
+                      {formatTimestamp(issue.detectedAt)}
+                    </div>
+                  </div>
```
[the remark this answers is not in the document]
Comment (rakshett) [reply src/components/Issues/LatestIssuesCard.tsx]: Used the Text from pattern fly

File: src/hooks/__tests__/useIssues.spec.ts [hunk 6]
```diff
@@ -0,0 +1,256 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { mockIssues } from '~/components/Issues/__data__/mock-issues-data';
+import { commonFetch } from '~/k8s/fetch';
+import { IssuesApiParams } from '~/types/issues';
+import { useIssues, useLatestIssues } from '../useIssues';
+
+jest.mock('~/k8s/fetch');
+
```
[the remark this answers is not in the document]
Comment (rakshett) [reply src/hooks/__tests__/useIssues.spec.ts]: This is been removed

File: src/components/Issues/LatestIssuesCard.tsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                    </Flex>
+                    <Text className="latest-issues-card__description">{issue.description}</Text>
+                    <Text component={TextVariants.small} className="latest-issues-card__timestamp">
+                      {formatTimestamp(issue.detectedAt)}
+                    </Text>
+                  </div>
+                </ListItem>
+              ))}
+            </List>
+            <div className="latest-issues-card__footer">
```
[the remark this answers is not in the document]
Comment (rakshett) [reply src/components/Issues/LatestIssuesCard.tsx]: Thanks @sahil143 I have remove the above and add the spacing for the button via the list by using margin bottom

File: src/components/Issues/__tests__/LatestIssuesCard.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+import { getErrorState } from '~/shared/utils/error-utils';
+import { renderWithQueryClientAndRouter } from '~/unit-test-utils/rendering-utils';
+import { LatestIssuesCard } from '../LatestIssuesCard';
+
+// Mock dependencies
+jest.mock('~/kite/kite-hooks', () => ({
+  useIssues: jest.fn(),
+}));
+
+jest.mock('~/shared/providers/Namespace', () => ({
```
Comment (rrosatti) [inline src/components/Issues/__tests__/LatestIssuesCard.spec.tsx]: you can use `mockUseNamespaceHook` from `src/unit-test-utils/mock-namespace.ts`

**Review by janaki29 — DISMISSED** [review]

**Review by StanislavJochman — APPROVED** [review]
LGTM

File: src/components/Issues/LatestIssuesCard.scss [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    margin-top: var(--pf-v5-global--spacer--sm);
+    margin-left: 0;
+  }
+
+  &__empty {
+    color: var(--pf-v5-global--Color--200);
+    font-style: italic;
+    text-align: center;
+    padding: var(--pf-v5-global--spacer--lg) 0;
+  }
```
Comment (sahil143) [inline src/components/Issues/LatestIssuesCard.scss]: These custom styles are not needed. Please use patternfly component for layout and Text component for styling

**Review by sahil143 — CHANGES_REQUESTED** [review]

**Review by testcara — APPROVED** [review]
LGTM

**Comment by rakshett** [comment]
> Code is good. I have the same question regarding the link. Both components are working as expected and as designed.
> 
> ps: Next please don't forget to fill `How to test` section.



> Code is good. I have the same question regarding the link. Both components are working as expected and as designed.
> 
> ps: Next please don't forget to fill `How to test` section.

I have added the 'How to test' section. Thanks for pointing this out

File: src/components/Issues/LatestIssuesCard.tsx [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+      <CardHeader>
+        <CardTitle component="h3">Latest issues</CardTitle>
+      </CardHeader>
+      <CardBody>
+        {hasIssues ? (
+          <>
+            <List isPlain className="latest-issues-card__list">
+              {issues.map((issue) => (
+                <ListItem key={issue.id} className="latest-issues-card__item">
+                  <div className="latest-issues-card__item-content">
```
Comment (sahil143) [inline src/components/Issues/LatestIssuesCard.tsx]: This seems like an extra div. Please remove it

File: src/components/Issues/LatestIssuesCard.scss [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+  &__icon-container {
+    flex-shrink: 0;
+    display: flex;
+    align-items: flex-start;
+  }
+
+  &__content {
+    min-width: 0;
+  }
```
Comment (sahil143) [inline src/components/Issues/LatestIssuesCard.scss]: This is not needed

File: src/components/Issues/LatestIssuesCard.tsx (hunk 10 above)
Comment (rakshett) [reply src/components/Issues/LatestIssuesCard.tsx]: Thanks! the CSS rule was not required and was creating unnecessary style conflicts.

File: src/components/Issues/LatestIssuesCard.scss [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    .pf-v5-c-list__item:not(:last-child) {
+      margin-bottom: var(--pf-v5-global--spacer--xl);
+      padding-bottom: var(--pf-v5-global--spacer--md);
+    }
+  }
+
+  &__item {
+    padding: 0;
+    margin-bottom: var(--pf-v5-global--spacer--lg);
+  }
```
[the remark this answers is not in the document]
Comment (rakshett) [reply src/components/Issues/LatestIssuesCard.scss]: I've removed the entire CSS override

File: src/components/Issues/LatestIssuesCard.scss [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    align-items: flex-start;
+  }
+
+  &__content {
+    min-width: 0;
+  }
+
+  &__icon {
+    font-size: var(--pf-v5-global--FontSize--md);
+  }
```
[the remark this answers is not in the document]
Comment (rakshett) [reply src/components/Issues/LatestIssuesCard.scss]: I have removed the css and updated as per above to use icon font size

**Review by testcara — APPROVED** [review]
LGTM

File: src/components/Issues/LatestIssuesCard.tsx [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+                        isInline
+                        component={(props) => <Link {...props} to={`/issues/${issue.id}`} />}
+                        style={{
+                          padding: 0,
+                          fontSize: 'inherit',
+                          fontWeight: 'bold',
+                          textAlign: 'left',
+                        }}
+                      >
+                        <Text component={TextVariants.h3} style={{ fontWeight: 'bold' }}>
```
Comment (rrosatti) [inline src/components/Issues/LatestIssuesCard.tsx]: nit: I'd update to use css style

**Review by sahil143 — APPROVED** [review]
